### DISCUSSION TITLE
Replace success response with error

### DISCUSSION THREAD

**Discussion by BigBadAlien** [klis87/redux-saga-requests#470] [Q&A]
In [OnError](https://github.com/klis87/redux-requests/blob/master/docusaurus/docs/tutorial/6-interceptors.md#onerror) interceptor we can replace error response with success. Do we have a way to do the opposite way for `OnSuccess`? Does it possible to replace success response with error?

**Comment by klis87** [comment]
@BigBadAlien not at the moment, it will be in the next version, with the new interceptors, we will have new `onResponse`, in which you will decide anything you want, error into success, success into error, even abort into success, whatever desired

Moreover, current `onError` and its family will be only for side-effects, they wont return anything, and will be called after `onResponse`. You might be interested in discussion https://github.com/klis87/redux-requests/issues/397 , new interceptors will solve more things, like not calling `getData` again for cached items in some cases, rehydrated data and so on.

**Comment by klis87** [comment]
@BigBadAlien what driver do you use though? perhaps you could achieve what you need on a driver level

**Reply by Antroll** [reply]
We are using the axios driver. Indeed, perhaps we can do this at the driver level, as suggested in https://github.com/klis87/redux-requests/issues/397#issuecomment-778606933.
